### ISSUE TITLE
Make handheld frag jumps more TFC-like™

### ISSUE THREAD

**Issue by squeek502** [fortressforever/fortressforever#41]
There's almost no variability in the trajectory you get from HHing nades in FF. In TFC, you had a lot more control (jump earlier for a flatter boost).

Also, the horizontal speed you get from HH frags in FF is much lower than it was in TFC (in TFC, as a spy, run + HH nade = ~1200 speed; in FF, you get ~600 speed if the nade goes off right as you jump, and ~900 speed if the nade goes off if you're on the downward part of your jump). There's also a huge difference in what happens if you HH without jumping between TFC and FF (in TFC, you go about 5 feet along the ground; in FF, you get popped up pretty high)

Referenced a little bit here: http://forums.fortress-forever.com/showpost.php?p=504571&postcount=13

**Comment by Largos** [comment]
Given how much damage they do, a horizontal boost would be more rewarding for all non-conc classes.  Would also help differentiate scout and medic from each other.  Supported.

**Comment by squeek502** [comment]
Did some testing using a custom TFC server plugin. Here's the resultant data:

| Velocity Before | Velocity After | Velocity Delta Length |
| --- | --- | --- |
| -333.065,-58.743,125.128 | -1239.192,-218.548,515.088 | 999.3360694751 |
| -230.221,302.97,236.328 | -735.157,967.452,797.816 | 1005.8655310547 |
| 0,0,0 | 0,0,951 | 951 |
| 0,0,196.328 | 0,0,1211.328 | 1015 |
| 113.123,-186.713,172.328 | 508.535,-839.353,826.64 | 1005.1934205356 |
| -388.906,164.596,220.328 | -1192.687,504.782,718.434 | 1004.937807923 |
| 362.817,-243.708,212.328 | 1098.42,-737.825,686.282 | 1004.9357080998 |
| -296.628,314.927,228.328 | -893.843,948.984,730.192 | 1005.2668849465 |
| 338.961,-232.455,197.128 | 1070.709,-734.273,666.642 | 1003.8524975433 |
| -39.372,283.965,173.128 | -183.006,1104.026,727.367 | 1000.1558072611 |
| 29.062,-299.655,189.128 | 109.021,-1124.006,757.125 | 1004.2751639322 |
| 45.706,250.873,220.328 | 177.975,976.94,905.534 | 1007.0628775236 |
| -48.377,-413.299,141.128 | -156.82,-1339.765,503.593 | 1000.7397312139 |
| -3.482,408.347,69.928 | -11.763,1380.378,285.775 | 995.7423197449 |
| 42.322,-358.393,172.328 | 146.21,-1238.146,641.864 | 1002.6076564883 |
| 54.978,-294.933,252.328 | 192.191,-1030.991,926.619 | 1007.6096165748 |
| -35.67,139.69,204.328 | -173.566,679.735,1046.919 | 1010.2591262256 |
| 341.839,-228.915,228.328 | 1056.66,-707.609,749.292 | 1005.7437531365 |
| 63.897,373.386,229.128 | 205.535,1201.058,780.778 | 1004.6989524868 |
| 345.759,-77.534,172.328 | 1206.868,-270.629,648.241 | 1002.6403006438 |
| -165.344,104.631,228.328 | -699.559,442.692,1015.042 | 1009.2521110912 |
| 196.979,-290.96,228.328 | 658.107,-972.08,807.113 | 1005.7611858732 |
| 0,0,133.128 | 0,0,1148.128 | 1015 |
| 0,0,21.928 | 0,0,1036.928 | 1015 |
| 0,0,-98.071 | 0,0,-1019.071 | 921 |
| 0,0,0 | 0,0,967 | 967 |
| 0,0,0 | 0,0,935 | 935 |
| -3.041,-171.476,0 | -19.548,-1102.675,0 | 931.3452950705 |
| 0,0,0 | 0,0,959 | 959 |
| 159.069,-310.834,181.128 | 555.762,-1086,679.322 | 1003.2172872519 |
| -37.15,306.048,204.328 | -135.426,1115.752,791.6 | 1005.0707148136 |
| 308.925,-303.067,236.328 | 925.954,-908.405,750.216 | 1005.6041754234 |
| -387.939,126.686,252.328 | -1183.312,386.428,811.502 | 1006.3596166227 |
| 359.343,-220.923,196.328 | 1119.761,-688.431,655.483 | 1003.8040539931 |
| 40.779,-297.215,213.128 | 149.171,-1087.142,826.074 | 1005.7018891844 |
| 358.148,-104.855,197.128 | 1191.602,-348.866,700.905 | 1003.9811770975 |
| -130.335,270.211,0 | -535.434,1110.069,0 | 932.4519612103 |
| 148.668,-260.865,196.328 | 553.981,-972.052,778.928 | 1004.7329679761 |
| -125.276,272.649,245.128 | -441.361,960.603,908.604 | 1006.6731524765 |
| 4.728,-148.964,189.128 | 23.69,-746.87,1002.048 | 1009.3017728509 |
| -4.315,296.908,52.328 | -18.372,1262.73,277.68 | 991.8635283329 |
| -100.575,-350.455,220.328 | -332.327,-1158.001,772.301 | 1005.2431170364 |
| -143.715,399.684,29.128 | -477.332,1327.509,147.27 | 993.0342720561 |
| -3.567,-428.322,-10.071 | -11.817,-1417.957,18.717 | 990.0879984471 |
| -93.396,362.57,51.528 | -336.837,1307.633,237.373 | 993.4515380606 |
| 8.512,-299.879,85.128 | 35.243,-1242.02,405.832 | 995.5878996141 |
| -46.188,296.542,181.128 | -175.067,1123.935,734.643 | 1003.7767821159 |
| -1.95,-259.986,133.128 | -8.474,-1128.427,630.335 | 1000.7233053677 |
| 17.556,299.495,109.128 | 71.388,1217.979,495.768 | 997.9986132656 |
| 44.135,-296.739,124.328 | 177.162,-1191.028,550.178 | 999.3984294314 |
| -36.879,192.657,61.928 | -211.258,1103.629,416.421 | 992.9467797793 |
| 17.416,-217.613,69.128 | 91.529,-1143.488,423.287 | 994.0653145921 |
| -341.296,272.612,252.328 | -1007.726,804.929,786.195 | 1006.2327300769 |
| 353.018,-255.49,157.128 | 1104.176,-799.136,536.329 | 1001.7897547295 |
| -160.508,399.375,228.328 | -484.929,1206.59,732.04 | 1005.2715157658 |
| 353.034,-237.587,236.328 | 1067.108,-718.145,756.369 | 1005.6243386678 |
| -402.503,124.469,236.328 | -1222.691,378.102,760.036 | 1005.6381671839 |
| 91.727,-404.003,196.328 | 258.926,-1275.447,661.715 | 1001.9756536493 |
| -376.963,187.766,252.328 | -1132.757,564.225,799.759 | 1006.2935202405 |

So, basically, it looks like it just adds ~1000 velocity in the direction of your current velocity. And if your velocity is (0,0,0), then it sends you straight up at ~950.

Should be fairly straightforward to implement.
